src/Server.php

```diff
@@ -93,21 +93,30 @@ public function run()
                 $response->writeHead(['Content-Type' => $request->getHeaders()['Accept'][0]]);
                 $response->end($body);
             } else {
-                $psr7Request = new ServerRequest(
-                    [],
-                    [],
-                    $request->getPath(),
-                    $request->getMethod(),
-                    (new Stream('php://input', 'w+')),
-                    $request->getHeaders(),
-                    $request->getHeader('cookie'),
-                    $request->getQueryParams()
-                );
-
-                $slimResponse = $this->slimInstance->process($psr7Request, new SlimResponse());
-                $response->writeHead($slimResponse->getStatusCode(), $slimResponse->getHeaders());
-                $slimResponse->getBody()->rewind();
-                $response->end($slimResponse->getBody()->getContents());
+                $stream = new Stream('php://memory', 'w+');
+
+                $request->on('data', function ($data) use ($stream) {
+                    $stream->write($data);
+                    $stream->rewind();
```

Comment:
@kelunik good spot! That was left in there in error. I have now fixed that in commit 458ebf9.